boards/arm/stm32f429i_disc1/stm32f429i_disc1_defconfig

```diff
@@ -29,6 +29,13 @@ CONFIG_MEMC=y
 # enable SDRAM
 CONFIG_MEMC_STM32_SDRAM=y
 
+# enable spi
+CONFIG_SPI=y
+CONFIG_SPI_STM32=y
+
+# enable display
+CONFIG_ILI9340=y
```

Comment:
You need to create an overlay in the sample, see examples https://github.com/zephyrproject-rtos/zephyr/tree/master/samples/display/lvgl/boards

Reply:
This is for the board since it has the display on it. It's not related to the sample.

Reply:
I've added sample overlay where it configures lvgl for the sample.

Reply:
This needs to be handled by the sample, see e.g. https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/display/lvgl/boards/reel_board.conf#L1

Reply:
> Just seen board already has ili9340 entry (not enabled, though). So all is needed are overlays in the sample both for Kconfig options and DT ili9340 node enable.

Why is it that we don't enable devices by default that are soldered on the board? This policy makes zero sense to me. It seems it only creates duplication of work for applications to have to add code to enable devices. Shouldn't the board be fully functional and configured out of the box when an application is built to run on it?

Reply:
> > Just seen board already has ili9340 entry (not enabled, though). So all is needed are overlays in the sample both for Kconfig options and DT ili9340 node enable.
> 
> Why is it that we don't enable devices by default that are soldered on the board? This policy makes zero sense to me. It seems it only creates duplication of work for applications to have to add code to enable devices. Shouldn't the board be fully functional and configured out of the box when an application is built to run on it?

Not all samples will need SPI or the display.

Reply:
@gmarull true. Is the policy though to not enable any board peripherals unless they are used? Should the board have all on board peripherals working even if hello world only uses UART?

Reply:
If all peripherals are enabled then binary size would be increased unnecessarily. While it is true that in general one tends to use all peripherals in a custom board, sample boards are used to evaluate many scenarios, and not all of them will need everything enabled.

Reply:
Actually boards that have displays do enable them by default but it is done via dts.

Eg for `mimxrt1050_evk` it is set to `OKAY`:
https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts#L102

Same for `reel_board`:
https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/reel_board/reel_board.dts#L54

Reply:
@gmarull @vanwinkeljan I think a good policy is to:
1. Enable everything available on the board in board defconfig
2. Have dts nodes configured with all necessary defaults to get everything working but set to status disabled in board dts
3. Use dts aliases for things like default display, default uart, default whatever
4. Enable what is needed in application dts overlay by referencing the aliases so that it can be done generically across all boards.  

This seems to provide the best balance between working configuration (so one doesn't have to pull hair out trying to figure out why devices that are on the board are not working) and small firmware size + maximum flexibility for applications (since they can overlay dts nodes with their own settings). 

What do you guys think? @erwango what do you think?